File: ios/MobilePassSDK/Model/BLEScanConfiguration.swift
//
//  BLEScanConfiguration.swift
//  MobilePassSDK
//
//  Created by Erinc Cakir on 28.02.2021.
//

import Foundation

public struct BLEScanConfiguration {
    var deviceList:         Dictionary<String, DeviceConnectionInfo>
    var dataUserId:         String
    var dataUserBarcode:    String
    var qrCodeId:           String
    var dataDirection:      Int
    var hardwareId:         String
    var deviceNumber:       Int
    var relayNumber:        Int
    var language:           Language
    
    init(devices: [ResponseAccessPointListTerminal], userId: String, userBarcode: String, qrCodeId: String, direction: Int, hardwareId: String, relayNumber: Int, language: Language) {
        self.dataUserId         = userId
        self.dataUserBarcode    = userBarcode
        self.dataDirection      = direction
        self.qrCodeId           = qrCodeId
        self.hardwareId         = hardwareId
        self.deviceNumber       = 0 // Default value
        self.relayNumber        = relayNumber
        self.language           = language
        
        self.deviceList = [:]
        
        for device in devices {
            if (device.i != nil && device.p != nil && !device.i!.isEmpty && !device.p!.isEmpty) {
                self.deviceList[device.i!.lowercased()] = DeviceConnectionInfo(deviceId: device.i!, publicKey: device.p!)
            }
        }
    }
}

File: android/mobilepasssdk/src/main/java/com/armongate/mobilepasssdk/model/BLEScanConfiguration.java
package com.armongate.mobilepasssdk.model;

import com.armongate.mobilepasssdk.enums.Language;
import com.armongate.mobilepasssdk.model.response.ResponseAccessPointListTerminal;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class BLEScanConfiguration {

    public Map<String, DeviceConnectionInfo> deviceList;
    public String dataUserId;
    public String dataUserBarcode;
    public String qrCodeId;
    public String hardwareId;
    public int direction;
    public int deviceNumber;
    public int relayNumber;
    public Language language;

    public BLEScanConfiguration(List<ResponseAccessPointListTerminal> devices, String userId, String userBarcode, String qrCodeId, String hardwareId, int direction, int relayNumber, Language language) {
        this.dataUserId         = userId;
        this.dataUserBarcode    = userBarcode;
        this.qrCodeId           = qrCodeId;
        this.hardwareId         = hardwareId;
        this.direction          = direction;
        this.deviceNumber       = 0; // Default value
        this.relayNumber        = relayNumber;
        this.language           = language;

        this.deviceList = new HashMap<>();

        for (ResponseAccessPointListTerminal device : devices) {
            if (device.i != null && !device.i.isEmpty() && device.p != null && !device.p.isEmpty()) {
                this.deviceList.put(device.i.toLowerCase(), new DeviceConnectionInfo(device.i, device.p));
            }
        }
    }

}
